client-react/src/pages/app/functions/function/function-log/FunctionLogAppInsightsDataLoader.tsx

```diff
@@ -256,12 +258,26 @@ const FunctionLogAppInsightsDataLoader: React.FC<FunctionLogAppInsightsDataLoade
     setAllLogEntries([]);
   };
 
+  const getShowFilteredLogsMessage = () => {
+    return (
+      !!Url.isFeatureFlagEnabled(CommonConstants.FeatureFlags.useNewFunctionLogsApi) &&
+      functionsRuntimeVersion === CommonConstants.FunctionsRuntimeVersions.four &&
+      !!liveLogsSessionId
+    );
+  };
```

Comment:
Consider using the `useCallback` hook since this relies on props:

```suggestion
  const getShowFilteredLogsMessage = useCallback(() => {
    return (
      !!Url.isFeatureFlagEnabled(CommonConstants.FeatureFlags.useNewFunctionLogsApi) &&

      functionsRuntimeVersion === CommonConstants.FunctionsRuntimeVersions.four &&
      !!liveLogsSessionId
    );
  }, [functionsRuntimeVersion, liveLogsSessionId];
```